### PR TITLE
devnet changes

### PR DESCRIPTION
**Pull request by DaMandal0rian**
This PR enables domains on all components and also sets a domain_id attribute.

### CODE REVIEW

File: network-primitives/bootstrap_node_provisioner.tf [hunk 1]
```diff
@@ -174,7 +174,7 @@ resource "null_resource" "start-boostrap-nodes" {
       # create docker compose file
       "sudo chmod +x /subspace/create_compose_file.sh",
       "sudo chmod +x /usr/bin/subspace",
-      "sudo /subspace/create_compose_file.sh ${var.bootstrap-node-config.reserved-only} ${length(local.bootstrap_nodes_ip_v4)} ${count.index}",
```
Comment (vedhavyas) [inline network-primitives/bootstrap_node_provisioner.tf]: `domain_id` is not required for all just rpc since that is where we run domains

**Review by vedhavyas — COMMENTED** [review]
We just need to update RPC nodes. rest can be left as is. 

Also need to update other resources apart from devnet. They should not give any terrform changes

**Comment by DaMandal0rian** [comment]
> We just need to update RPC nodes. rest can be left as is.
> 
> Also need to update other resources apart from devnet. They should not give any terrform changes

@vedhavyas this branch is for devnet only, will open a PR for the gemini-3d terraform

File: network-primitives/variables.tf [hunk 2]
```diff
@@ -32,6 +32,7 @@ variable "full-node-config" {
     reserved-only       = bool
     prune               = bool
     node-dsn-port       = number
+    domain-id           = number
```
Comment (vedhavyas) [inline network-primitives/variables.tf]: forgot to remove these ?

File: resources/devnet/variables.tf [hunk 3]
```diff
@@ -30,3 +30,9 @@ variable "farmer-reward-address" {
   description = "Farmer's reward address"
   type        = string
 }
+
+variable "domain_id" {
+  description = "Domain ID"
+  type        = number
+  default     = 1
+}
```
Comment (vedhavyas) [inline resources/devnet/variables.tf]: new line pls

**Review by vedhavyas — COMMENTED** [review]
Make sense. Some nits and there are some forgotten changes that are not required

File: network-primitives/variables.tf (hunk 2 above)
Comment (DaMandal0rian) [reply network-primitives/variables.tf]: thanks, good catch!

Comment (DaMandal0rian) [reply network-primitives/variables.tf]: i actually removed them, and my git work tree is clean. Need to check why it wasn't staged. :/

**Pull request closed without merging**